### ISSUE TITLE
Discovery Go URL falling back on generic information extractor

### ISSUE THREAD

**Issue by cas4** [ytdl-org/youtube-dl#21861]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2019.07.16**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
youtube-dl.exe --verbose --hls-prefer-native --ap-mso Comcast_SSO --ap-username PRIVATE --ap-password PRIVATE https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--hls-prefer-native', '--ap-mso', 'Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', 'https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2019.07.16
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.17134
[debug] exe versions: ffmpeg N-61620-ge555e1b, ffprobe git-2017-12-23-d02289c
[debug] Proxy map: {}
[generic] buckers-and-brawlers-2: Requesting header
WARNING: Falling back on generic information extractor.
[generic] buckers-and-brawlers-2: Downloading webpage
[generic] buckers-and-brawlers-2: Extracting information
ERROR: Unsupported URL: https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpaxmfe_l6\build\youtube_dl\YoutubeDL.py", line 796, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpaxmfe_l6\build\youtube_dl\extractor\common.py", line 530, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpaxmfe_l6\build\youtube_dl\extractor\generic.py", line 3333, in _real_extract
youtube_dl.utils.UnsupportedError: Unsupported URL: https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2

youtube-dl.exe --verbose --hls-prefer-native --cookies ./cookies_discovery.txt https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--hls-prefer-native', '--cookies', './cookies_discovery.txt', 'https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2019.07.16
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.17134
[debug] exe versions: ffmpeg N-61620-ge555e1b, ffprobe git-2017-12-23-d02289c
[debug] Proxy map: {}
[generic] buckers-and-brawlers-2: Requesting header
WARNING: Falling back on generic information extractor.
[generic] buckers-and-brawlers-2: Downloading webpage
[generic] buckers-and-brawlers-2: Extracting information
ERROR: Unsupported URL: https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpaxmfe_l6\build\youtube_dl\YoutubeDL.py", line 796, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpaxmfe_l6\build\youtube_dl\extractor\common.py", line 530, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpaxmfe_l6\build\youtube_dl\extractor\generic.py", line 3333, in _real_extract
youtube_dl.utils.UnsupportedError: Unsupported URL: https://go.discovery.com/tv-shows/battlebots/full-episodes/buckers-and-brawlers-2

```

## Description

When passing a "go.discovery.com" URL, it falls back to the generic information extractor after requesting the header.  I opened ticket #21843 and it was closed with "You must provide account credentials.", however as I demonstrated above, I tried with both the cookies file (worked up until ~7/7/2019) and I also tried using the --ap-mso switch to pass in my credentials.  Both failed with the same error message.  This does not appear to be a permissions issue, but something appears to have changed in the discovery.com website.

**Comment by cas4** [comment]
Can I please get an explanation as to why these cases are being closed? #21843 was closed with the comment account credentials needed and they were through the cookie file. I opened this case showing the output using the ap-mso switch the show it wasn't an account login problem and pointing out that the discovery exctracter was not used for a discovery url.  Any insight into this problem would be appreciated.

**Comment by Diegus83** [comment]
The problem is that Discovery changed the URLs to "go.discovery.com" and those are not picked up by the extractor. It has been fixed by this commit
https://github.com/ytdl-org/youtube-dl/commit/3b446ab3519948980630e3328b971385826ffba8

You can run the current code in developer mode or wait for the next release which will probably include the fix.
